Registering agent improvement

dkempny opened the pull request:
Hello Team,
this PR closes #2320 


## Description

The modified files propose a new approach to present Wazuh agent registration process.


## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [ ] Used impersonal speech. 
- [x] Used uppercase only on nouns. 

All the best,
Daria

davidfmiranda commented on source/user-manual/registering/restful-api-register.rst:
```diff
@@ -1,29 +1,202 @@
+             }
+            }
+
+    More about API credentials and HTTPS support can be found on :ref:`Wazuh API configuration<api_configuration>`.
```
```suggestion
    More information about API credentials and HTTPS support can be found on :ref:`Wazuh API configuration<api_configuration>`.
```

dkempny replied:
Done

davidfmiranda commented on source/user-manual/registering/restful-api-register.rst:
```diff
@@ -1,29 +1,202 @@
+
+    More about API credentials and HTTPS support can be found on :ref:`Wazuh API configuration<api_configuration>`.
+
+   2. Import the registration key to the agent using ``manage_agents`` program. Replave the agent's registration key:
```
```suggestion
   2. Import the registration key to the agent using ``manage_agents`` program. Replace the agent's registration key:
```

dkempny replied:
Done

davidfmiranda commented on source/user-manual/registering/restful-api-register.rst:
```diff
@@ -1,29 +1,202 @@
+             }
+            }
+
+    More about API credentials and HTTPS support can be found on :ref:`Wazuh API configuration<api_configuration>`.
```
```suggestion
    More information about API credentials and HTTPS support can be found on :ref:`Wazuh API configuration<api_configuration>`.
```

dkempny replied:
Done

davidfmiranda commented on source/user-manual/registering/using-command-line.rst:
```diff
@@ -1,18 +1,154 @@
 ===============================================
 
-This method consists of registering the agents manually in the Wazuh manager using ``manage_agents`` binary, extracting the key and inserting it in the Wazuh agents using ``manage_agents`` too.
+This method consists of registering the Wazuh agent manually with the Wazuh manager using ``manage_agents`` program, extracting the registration key and inserting it in the agent.
```
```suggestion
This method consists on registering the Wazuh agent manually with the Wazuh manager using ``manage_agents`` program, extracting the registration key and inserting it in the agent.
```

dkempny replied:
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,41 @@ Registering agents
   :description: Learn more about the different methods that can be used to register agents against the Wazuh manager.
 
-In this section, the registration process is described, and more specifically the different methods that users can use to register agents in the Wazuh manager.
+The Wazuh agent registration process enables Agent-Manager communication, which requires a pre-shared key to be available on both sides. This key is used to establish an encrypted and authenticated communication channel by using `AES algorithm <https://wazuh.com/blog/benefits-of-using-aes-in-our-communications/>`_, allowing the manager to receive data collected by the agent.
```
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,41 @@ Registering agents
+.. note::
 
-The registration service requires an SSL certificate on the manager in order to work. This certificate will be automatically generated by the package during the installation if the ``openssl`` package is installed. This certificate is stored in ``/var/ossec/etc/sslmanager.cert``.
+    Note that if the agent was deployed using :ref:`Deployment Variables <deployment_variables>`, :ref:`Deployed with Ansible <wazuh_ansible>` or :ref:`Deployed with Puppet <wazuh_puppet>`, the registration process is different and described in corresponding sections of the documentation.
```
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,41 @@ Registering agents
++------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Registration method                                                                                        | Description                                                                                                                                                                                                                    |
++============================================================================================================+================================================================================================================================================================================================================================+
+| :ref:`Using command line (CLI) <using-command-line>`                                                       | Manual registeration using ``manage_agents`` program. Requires extracting the registration key from the manager and inserting it in the agent.                                                                                 |
```
Section removed in the new version

dkempny commented on source/user-manual/registering/registering-agents-troubleshooting.rst:
```diff
@@ -0,0 +1,58 @@
+Agent registers successfully but disappears after another agent is being registered.
+------------------------------------------------------------------------------------
+
+By default, the manager attaches an agent to the visible IP of the agent. If the manager or the agents are, for example, behind a NAT, the agent should be registered with IP Address ``any``. That way ``any`` works as ``0.0.0.0/0``.
```
Done

dkempny commented:
New proposed structure added to the branch.

alberpilot commented on source/_templates/registrations/linux/start_agent.rst:
```diff
@@ -0,0 +1,17 @@
+.. Copyright (C) 2020 Wazuh, Inc.
```
Those restart agent's methods not represent how to restart Wazuh agents in AIX, Solaris 10/11 and HP-UX.

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,104 @@ Registering agents
+.. _simple-registration-service:
 
-.. code-block:: console
+Registering agent using Simple Registration Service
```
Use the lowercase

dkempny replied:
Done

alberpilot commented on source/_templates/registrations/common/set_agent_name.rst:
```diff
@@ -0,0 +1,7 @@
+
+.. note::
+
+ If the new agent’s name is not provided, it is set automatically using hostname. To specify the agent's name add ``-A <agent_name>`` to the command below.
```
This comment must not be a Note. Please include it after the command as normal text.

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,104 @@ Registering agents
+  The agent registration can be adjusted by using different :ref:`agent-auth` options.
+
+
+The above method is the simplest way of registering the agent. There are also available :ref:`other registration methods <other_registration_methods>`.
```
I think that it would be great if we have here a table summarizing all the Other registration methods.

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,104 @@ Registering agents
-    password-authorization-registration-service
-    manager-verification/host-verification-registration
-    restful-api-register
+    other-registration-methods
```
And I think that every other registration method must have a dedicated single page in order to avoid the user think that all the page steps must be completed in a registration alternative process.

dkempny replied:
Done

dkempny commented:
1. I have an issue with a **tab with a Unix command** for the agent's restart. The tab gets a scroll suggesting that the command is too long. 
2. I removed the **troubleshooting** section for now as its content will land on another page. I need to review the content as well as it might not be correct with the v3.12.0.
3. I added the new **opening section paragraph**, but I have doughts if it is correct with all registration methods.

alberpilot commented on source/_templates/registrations/linux/start_agent.rst:
```diff
@@ -0,0 +1,24 @@
+
+   # service wazuh-agent restart
+
+ .. group-tab:: For Unix host
```
I don't understand this template. Start agent or restart agent?

alberpilot replied:
Anyway, the tabs titles must be: Systemd, SysV init, Other Unix based OS

dkempny replied:
Done

alberpilot commented on source/_templates/registrations/windows/start_agent.rst:
```diff
@@ -0,0 +1,18 @@
+
+  .. code-block:: console
+
+   # Restart-Service -Name wazuh
```
Same, start or restart? not the same

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,118 @@ Registering agents
 
-Prerequisites
--------------
+ - 1515/TCP: Service used for agents registration, to obtain a unique key. The communication is done over TLS. The obtained unique key (one per agent) is used later to authenticate with the Wazuh service (1514/TCP-UDP) and to encrypt traffic.
```
Uppercase after `:` is not needed.

alberpilot replied:
is the `,` well used?

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,118 @@ Registering agents
   :description: Learn more about the different methods that can be used to register agents against the Wazuh manager.
 
-In this section, the registration process is described, and more specifically the different methods that users can use to register agents in the Wazuh manager.
+Wazuh agents communicate with the Wazuh server using outbound TCP/UDP connections to two different services:
```
It is not said that 1515 and 1514 are default ports.

alberpilot replied:
Wazuh agent communicates with the Wazuh manager

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -8,56 +8,118 @@ Registering agents
+.. note::
 
-If you already have a certificate, you can use them just by copying them into the same path:
+    If the agent was deployed using :ref:`Deployment variables <deployment_variables>`, :ref:`Deployed with Ansible <wazuh_ansible>` or :ref:`Deployed with Puppet <wazuh_puppet>`, the registration process is different and described in their corresponding sections of the documentation.
```
Can we explain here that if the deployment was done with variables, this section is completely unuseful?

dkempny replied:
Done, hopefully it is better now.

alberpilot commented on source/user-manual/registering/command-line-registration.rst:
```diff
@@ -0,0 +1,154 @@
+            Confirm adding it?(y/n): y
+            Added.
+
+   2. To enable the communication with the manager, edit the agent's ``/var/ossec/etc/ossec.conf`` configuration file:
```
Let's try to avoid "manager" term. The correct term is "Wazuh manager"

dkempny replied:
Done for the whole section.

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
 
-Prerequisites
--------------
+ - 1515/TCP: service used for Wazuh agents registration to obtain a unique key. The communication is done over TLS. The obtained unique key (one per Wazuh agent) is used later to authenticate with the Wazuh service (1514/TCP-UDP) and to encrypt traffic.
```
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
 
-If you already have a certificate, you can use them just by copying them into the same path:
+    This documentation section can be skipped if the Wazuh agent was deployed using :ref:`Deployment variables <deployment_variables>`, :ref:`Deployed with Ansible <wazuh_ansible>` or :ref:`Deployed with Puppet <wazuh_puppet>`. In those cases, the registration process is different and described in their corresponding sections of the documentation.
+    If the Wazuh runs in the cluster mode, please refer to the :ref:`Configuring a cluster section <load_balancer>` to get more information about the registration process in the cluster.
```
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
 
-Available registration methods
-------------------------------
+   Open a session in the Linux/Unix Wazuh agent's host as a ``root`` user.
```
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
+   Open a session in the Linux/Unix Wazuh agent's host as a ``root`` user.
 
-You can use the method that better suits with your need. We recommend: :ref:`simple-registration-service`.
+   1. To register the Wazuh agent, run the ``agent-auth`` program, using the Wazuh manager’s IP address:
```
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
+  :description: Learn more about the different methods that can be used to register Wazuh agents against the Wazuh manager.
 
-In this section, the registration process is described, and more specifically the different methods that users can use to register agents in the Wazuh manager.
+By default, Wazuh agents communicate with the Wazuh manager using outbound TCP/UDP connections to two different services:
```
I think that we need to improve this part. 
`two different services` and then `1515/TCP: service used for Wazuh agents`. This is not a service. `1515/TCP` is the used port `1515` with protocol `TCP` that `ossec-auth` use by default for listening register requests.

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
++-----------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Registration method                                                                                 | Description                                                                                                                                                  |
++=====================================================================================================+==============================================================================================================================================================+
+| :ref:`Using command line (CLI) <command-line-registration>`                                         | Manual registeration using ``manage_agents`` utility. Requires extracting the registration key from the Wazuh manager and inserting it in the Wazuh agent.   |
```
Inserting it manually in the Wazuh agent.

dkempny replied:
Done

alberpilot commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
++=====================================================================================================+==============================================================================================================================================================+
+| :ref:`Using command line (CLI) <command-line-registration>`                                         | Manual registeration using ``manage_agents`` utility. Requires extracting the registration key from the Wazuh manager and inserting it in the Wazuh agent.   |
++-----------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| :ref:`Using Wazuh API <restful-api-registration>`                                                   | Uses a simple Wazuh API request from any host. Requires adding returned registration key to the Wazuh agent using ``manage_agents`` utility.                 |
```
Same

dkempny replied:
Done

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
+  :description: Learn more about the different methods that can be used to register Wazuh agents against the Wazuh manager.
 
-In this section, the registration process is described, and more specifically the different methods that users can use to register agents in the Wazuh manager.
+By default, Wazuh agents communicate with the Wazuh manager using outbound TCP/UDP connections to two different services:
```
Not yet done. I didn't produce a clear enough version of this paragraph.

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
+
+    .. include:: ../../_templates/registrations/common/set_agent_name.rst
+
+   2. To enable the communication with the Wazuh manager, edit the Wazuh agent's  ``/var/ossec/etc/ossec.conf`` configuration file:
```
Done.

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,127 @@
+
+  The Wazuh agent registration can be adjusted by using different :ref:`agent-auth` options.
+
+The above method is the simplest way of registering the Wazuh agents. There are also available other registration methods:
```
Done.

dkempny commented:
I am still tweaking the opening paragraph section so the information can be more clear.

dkempny commented on source/user-manual/registering/index.rst:
```diff
@@ -1,63 +1,125 @@
+  :description: Learn more about the different methods that can be used to register Wazuh agents against the Wazuh manager.
 
-In this section, the registration process is described, and more specifically the different methods that users can use to register agents in the Wazuh manager.
+The security event data collection from the Wazuh agent requires to enable the communication with the Wazuh manager. The Wazuh manager must know which Wazuh agent is sending the security events and if it is authorized. This step is called as Wazuh agent registration and it can be done by using the ``registration service``. Using the port 1515 and TCP protocol, the Wazuh manager will attend the re ...
```
Done

dkempny commented on source/_templates/registrations/windows/installation_directory.rst:
```diff
@@ -0,0 +1,12 @@
+The Wazuh agent's installation directory depends on the architecture of the host:
+
+ - ``C:\Program Files (x86)\ossec-agent`` for ``x86_64`` hosts.
+ - ``C:\Program Files\ossec-agent`` for ``x64`` hosts.
```
Done
